### PR TITLE
fix(nuxt): infer `useFetch` method when generic is passed

### LINKED ISSUE #19077: PUT, PATCH, and DELETE requests
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.18.1`
- Nuxt Version:     `3.2.0`
- Nitro Version:    `2.2.1`
- Package Manager:  `npm@`
- Builder:          `vite`
- User Config:      `runtimeConfig`, `modules`, `tailwindcss`, `nitro`, `app`, `typescript`, `routeRules`
- Runtime Modules:  `@nuxtjs/tailwindcss@6.4.0`, `nuxt-icon@0.2.11`
- Build Modules:    `-`
------------------------------

### Reproduction

I have a variable defined like `const method = someBooleanCheck ? "PATCH": "PUT";`

I use it as `await useFetch("/api/fileName", { method: method, body: { some: data} });`

[https://stackblitz.com/edit/nuxt-patch-gjclzp](https://stackblitz.com/edit/nuxt-patch-gjclzp)

### Describe the bug

This causes a TS compiler error `Type '"PATCH" | "PUT"' is not assignable to type '"get" | "GET" | "POST" | "post" | Ref<"get" | "GET" | "POST" | "post" | undefined> | undefined'.`

### Additional context

I raised a discussions #19065 because I use the request keyword in the file name - `/server/api/fileName.patch.ts`.

Everything works as expected when I silence the compiler error with `// @ts-ignore`, but this is in the development.
I don't know if I should submit it to production like this or refactor the call to POST with some additional checks.

FYI
The project lives in a Firebase environment.

### Logs

_No response_

### PR DESCRIPTION
**Pull request by Gypsophyllite**
### 🔗 Linked issue

resolves #19077 

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

I provided a [workaround](https://github.com/nuxt/nuxt/issues/19077#issuecomment-1434008688) before, but this issue doesn't get solved. 

I find that just change the method type `'get' extends AvailableRouterMethod<ReqT> ? 'get' : AvailableRouterMethod<ReqT>` into `AvailableRouterMethod<ReqT>`, and all works fine as I see

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by danielroe — COMMENTED** [review]
Would you be able to add a test case in `test/fixtures/basic/types.ts` that was failing before this PR?

**Comment by Gypsophyllite** [comment]
@danielroe I added one, btw, setting the return type and request method explicitly will easily reproduce.

**Comment by danielroe** [comment]
Thanks! I've gone ahead and pushed a failing test to show why the type is written as is.

File: packages/nuxt/src/app/composables/fetch.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export function useFetch<
   ResT = void,
   ErrorT = FetchError,
   ReqT extends NitroFetchRequest = NitroFetchRequest,
-  Method extends AvailableRouterMethod<ReqT> = 'get' extends AvailableRouterMethod<ReqT> ? 'get' : AvailableRouterMethod<ReqT>,
+  Method extends AvailableRouterMethod<ReqT> = AvailableRouterMethod<ReqT>,
```
Comment (danielroe) [inline packages/nuxt/src/app/composables/fetch.ts:34]: What about this?

```suggestion
  Method extends AvailableRouterMethod<ReqT> = ResT extends void ? 'get' extends AvailableRouterMethod<ReqT> ? 'get' : AvailableRouterMethod<ReqT> : AvailableRouterMethod<ReqT>,
```

Comment (Gypsophyllite) [reply packages/nuxt/src/app/composables/fetch.ts]: I also find the reason of broken type when passing uppercase method, maybe its a nitro type error in `fetch.ts`.
Another PR in [nitro](https://github.com/unjs/nitro/pull/1247)

Comment (danielroe) [reply packages/nuxt/src/app/composables/fetch.ts]: Nice work - thanks! I think we can go ahead and merge this PR (as it fixes _one_ issue without introducing any regressions) and then update test once nitro PR is merged..

**Review by danielroe — APPROVED** [review]

**Pull request merged**